### PR TITLE
Add support for *string in redisx

### PR DESCRIPTION
**Pull request by domenipavec**
Add support for *string data receiver in redisx
Using `*string` receiver for data from redis enables distinguishing between empty string values and `nil` (e.g. stored empty string vs. cache miss).

### CODE REVIEW

File: redis/cache/redisx/request.go [hunk 1]
```diff
@@ -249,18 +250,19 @@ func (r Request) setStructValue(dst reflect.Value, src reflect.Value) error {
 func (r Request) convertAndSetByteSlice(dst reflect.Value, src reflect.Value) error {
 	asBytes, _ := src.Interface().([]byte)
 	asStr := string(asBytes)
-	switch dst.Kind() {
-	case reflect.Int64:
```
Comment (rbietin) [inline redis/cache/redisx/request.go:253]: Why did you change this switch?

Comment (domenipavec) [reply redis/cache/redisx/request.go:253]: @rbietin Because in my case I have to check `dst.Type()` and not `dst.Kind()` that is checked in other cases.

**Review by rbietin — APPROVED** [review]

**Review by kylelemons — CHANGES_REQUESTED** [review]

File: redis/cache/redisx/request.go (hunk 1 above)
Comment (domenipavec) [reply redis/cache/redisx/request.go]: @fishy @kylelemons Ok, changed back to using `switch`.

File: redis/cache/redisx/request_benchmark_test.go [hunk 2]
```diff
@@ -11,6 +11,8 @@ import (
 // BenchmarkSetValue
 // BenchmarkSetValue/string
 // BenchmarkSetValue/string-12         	14946445	        67.1 ns/op
+// BenchmarkSetValue/*string
+// BenchmarkSetValue/*string-12	         6187052	        169 ns/op
```
Comment (fishy) [inline redis/cache/redisx/request_benchmark_test.go:15]: 🔕 this looks a lot slower than others (for example, `string`). it's probably due to your machine being slower than the original source of the benchmark results, you should replace the whole result, not only add the new one you added.

**Review by fishy — APPROVED** [review]

**Review by konradreiche — APPROVED** [review]

File: redis/cache/redisx/request_benchmark_test.go (hunk 2 above)
Comment (domenipavec) [reply redis/cache/redisx/request_benchmark_test.go:15]: The values for others are comparable on my machine. This is slower than the `string` benchmark, because it's going from `[]byte` to `*string` so it's doing type conversion, while `string` benchmark is just doing the assignment from string to string. If I change the `string` benchmark to go from `[]byte` it's just as slow.

File: redis/cache/redisx/request.go [hunk 3]
```diff
@@ -261,9 +262,13 @@ func (r Request) convertAndSetByteSlice(dst reflect.Value, src reflect.Value) er
...
-			Cmd:               r.Cmd,
-			ResponseInputType: dst.Type(),
+		if dst.Type() == stringPtrT {
+			dst.Set(reflect.ValueOf(&asStr))
+		} else {
+			return &ResponseInputTypeError{
+				Cmd:               r.Cmd,
+				ResponseInputType: dst.Type(),
+			}
 		}
```
Comment (kylelemons) [inline redis/cache/redisx/request.go:272]: 🔕 this can be a switch too

```suggestion
		switch dst.Type() {
		case stringPtrT:
			dst.Set(reflect.ValueOf(&asStr))
		default:
			return &ResponseInputTypeError{
				Cmd:               r.Cmd,
				ResponseInputType: dst.Type(),
			}
		}
```

**Review by kylelemons — APPROVED** [review]

**Review by pacejackson — CHANGES_REQUESTED** [review]
The change seems fine but it should probably also support *int64, it seems like it would be confusing to support this for the string conversion but not the integer one. 

There’s also documentation here that needs to be updated to reflect the new types you can use. https://github.com/reddit/baseplate.go/blob/70ef150cf4a06005b64806bcf3c62bc73b128710/redis/cache/redisx/syncx.go#L27

File: redis/cache/redisx/request_benchmark_test.go (hunk 2 above)
Comment (domenipavec) [reply redis/cache/redisx/request_benchmark_test.go]: I've modified the benchmarks to include source and destination types so the results are easier to interpret.

**Comment by domenipavec** [comment]
@pacejackson Added support for `*int64` and updated documentation.
